js/src/components/shipping-rate-section/flat-shipping-rates-input-cards.js

```diff
@@ -2,22 +2,78 @@
  * Internal dependencies
  */
 import isNonFreeFlatShippingRate from '.~/utils/isNonFreeFlatShippingRate';
-import FreeShippingCards from './free-shipping-cards';
 import EstimatedShippingRatesCard from './estimated-shipping-rates-card';
+import OfferFreeShippingCard from './offer-free-shipping-card';
+import MinimumOrderCard from './minimum-order-card';
 
 const FlatShippingRatesInputCards = ( props ) => {
-	const { audienceCountries, value, onChange = () => {} } = props;
-	const displayFreeShippingCards = value.some( isNonFreeFlatShippingRate );
+	const { audienceCountries, formProps } = props;
+	const { getInputProps, values, setValue } = formProps;
+	const displayFreeShippingCards = values.shipping_country_rates.some(
+		isNonFreeFlatShippingRate
+	);
+
+	const handleShippingRatesChange = ( onChange ) => ( newShippingRates ) => {
+		/**
+		 * If all the shipping rates are free shipping,
+		 * we set the offer_free_shipping to undefined,
+		 * so that when users add a non-free shipping rate,
+		 * they would need to choose "yes" / "no" for offer_free_shipping.
+		 */
+		if ( ! newShippingRates.some( isNonFreeFlatShippingRate ) ) {
+			setValue( 'offer_free_shipping', undefined );
+		}
+
+		onChange( newShippingRates );
+	};
+
+	const handleOfferFreeShippingChange = ( onChange ) => (
```

Comment:
💅  Same naming concern as the above `handleShippingRatesChange`.

Reply:
Renamed to `getOfferFreeShippingChangeHandler` in [`6f85c1d` (#1302)](https://github.com/woocommerce/google-listings-and-ads/pull/1302/commits/6f85c1d8c8c8863335ff5432a17838d9badaffa5).